### PR TITLE
Replace PEP8 by pycodestyle

### LINKED ISSUE #1132: Replace pep8 by pycodestyle in static code analysis
The `pep8` tool has been renamed to `pycodestyle`:
```
pep8 has been renamed to pycodestyle (GitHub issue #466)
Use of the pep8 tool will be removed in a future release.
Please install and use `pycodestyle` instead.
```
We should update our static code analysis scripts.

### LINKED ISSUE #1682: Fix code formatting for all files in the repository once and for all
It seems that not all files in the repository currently conform to our static code style guidelines and people are sometimes hit by this when they edit a file, pertaining files are then checked and unrelated code locations evoke errors.

In order to ease future development, all files in the repository should again be checked for conformity (PEP8, clang-format) and committed.

### PR DESCRIPTION
**Pull request by pnbabu**
Fixes #1132, fixes #1682.

- Replace PEP8 with pycodestyle executable 
- Rename some variables related to Travis CI to Github Actions.
- Remove some unused code related to Travis CI

### CODE REVIEW

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:112]:
```suggestion
                                           the NEST CI build and test environment.
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:118]:
```suggestion
                                           the NEST CI build and test environment.
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:79]: Let's make our life easier the next time we change CI systems ;-)

```suggestion
coding  style  guidelines.  The  checks are performed the same way as in the NEST CI
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:285]: I just fixed some of these W504 warnings (line break after binary operator) in a branch of mine and in return got a bunch of W503 (line break *before* binary operator) :disappointed:

As I think the readability of those code locations depends more on context and a general rule will always fail in some cases, I suggest to ignore both: 


```suggestion
  $PEP8 --ignore="E121,E501,W503,W504" ./extras/parse_build_log.py || error_exit "Failed to verify the pycodestyle installation. Executable: $PEP8"
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:306]:
```suggestion
RUNS_ON_CI=false
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:310]:
```suggestion
IGNORE_MSG_CPPCHECK=false      # the CI build environment.
```

File: extras/check_code_style.sh
Comment (jougs) [inline extras/check_code_style.sh:314]:
```suggestion
./extras/static_code_analysis.sh "$RUNS_ON_CI" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```

File: extras/gha_build.sh
Comment (jougs) [inline extras/gha_build.sh:23]:
```suggestion
# This shell script is part of the NEST CI build and test environment.
```

File: extras/gha_build.sh
Comment (jougs) [inline extras/gha_build.sh:115]:
```suggestion
    # will cause the CI build to fail or are ignored.
```

File: extras/gha_build.sh
Comment (jougs) [inline extras/gha_build.sh:122]:
```suggestion
    # The script is called within the CI environment and thus can not be run incremental.
    RUNS_ON_CI=true
```

File: extras/gha_build.sh
Comment (jougs) [inline extras/gha_build.sh:126]:
```suggestion
    ./extras/static_code_analysis.sh "$RUNS_ON_CI" "$INCREMENTAL" "$file_names" "$NEST_VPATH" \
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:23]:
```suggestion
# This shell script is part of the NEST CI build and test environment.
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:24]: I suggest to rename `extras/gha_build.sh` to `ci_build.sh` and change all references to it accordingly.

```suggestion
# It performs the static code analysis and is invoked by 'ci_build.sh'.
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:35]:
```suggestion
RUNS_ON_CI=${1}               # true or false, indicating whether the script is executed on the CI or not.
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:53]: Please also add W503 and W504 here. See [above](https://github.com/nest/nest-simulator/pull/2105/files#diff-cb3c61b3c773983f77fd537c8923e562835078815e095a308a10d176cfe54a1bR285) for why.

I also dislike it a bit that the verification above uses its own set of ignores and does not use the list defined here. How complicated would it be to do so?

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:68]:
```suggestion
# The format of the message depends on whether the script is executed on CI or not.
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:71]:
```suggestion
  if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:101]:
```suggestion
# static code analysis error messages will cause the CI build to fail or are ignored.
if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:111]:
```suggestion
  if $RUNS_ON_CI && $IGNORE_MSG_PYCODESTYLE; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:140]:
```suggestion
  if ! $RUNS_ON_CI && $INCREMENTAL; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:145]:
```suggestion
  if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:155]:
```suggestion
      if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:173]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:190]:
```suggestion
            if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:200]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:222]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:256]:
```suggestion
        if $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:285]:
```suggestion
    if ! $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:314]:
```suggestion
    if ! $RUNS_ON_CI; then
```

File: extras/static_code_analysis.sh
Comment (jougs) [inline extras/static_code_analysis.sh:320]:
```suggestion
  if ! $RUNS_ON_CI; then
```

**Review by jougs — CHANGES_REQUESTED** [review]
Many thanks for your contribution. I have added a couple of comments and suggestions, most of which you can probably just batch commit.

File: extras/static_code_analysis.sh
Comment (pnbabu) [reply extras/static_code_analysis.sh:53]: You are right. I will make a list of error and warning codes and pass them to this script.

File: extras/static_code_analysis.sh
Comment (pnbabu) [reply extras/static_code_analysis.sh]: Done

Comment (jougs) [reply extras/static_code_analysis.sh]: Now, the script is called just `ci.sh`, which I find a bit too brief. Can you please make it `ci_build.sh`? Or is there a reason for your shorter choice?

Comment (pnbabu) [reply extras/static_code_analysis.sh]: I somehow read your comment wrongly and assumed it as `ci.sh`. Sorry, my bad. I have fixed this correctly now as `ci_build.sh`.

**Comment by jougs** [comment]
@pnbabu: I have sent you https://github.com/pnbabu/nest-simulator/pull/1, as I thought this would fit nicely here. If you decide to merge it, please also make to mark #1682 as being fixed by this PR.

**Comment by jougs** [comment]
Fixes #1682.

**Review by jougs — APPROVED** [review]
LGTM! Merging without second review, once the CI is happy.

**Pull request merged**
